go-controller/pkg/ovn/egressip.go

```diff
@@ -1237,82 +1269,60 @@ func (oc *Controller) deleteDefaultNoRerouteNodePolicies(v4NodeAddr, v6NodeAddr
 	return nil
 }
 
-func createNATRule(podIPs []net.IP, status egressipv1.EgressIPStatusItem, egressIPName string) error {
+func buildSNAT(podIP net.IP, status egressipv1.EgressIPStatusItem, egressIPName string) (*nbdb.NAT, error) {
```

Comment:
This function should either not take an `egressipv1.EgressIPStatusItem` and be moved to the utils package, or change it's name to reflect that it's solely building SNATs for the purpose of egress IP. I prefer the first option.

Reply:
You already have a `BuildRouterSNAT` and `BuildRouterDNATAndSNAT`, maybe you can just have the latter do OR instead and keep only one function?